Sources/Jaeger/Sampling/ProbabilisticSampler.swift

```diff
@@ -0,0 +1,32 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Jaeger Client Swift open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Jaeger Client Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import W3CTraceContext
+
+public struct ProbabilisticSampler: Sampler {
+    private let probability: Double
+    private let lowUpperBound: UInt64
+
+    public init(probability: Double = 0.001) {
+        precondition(probability > 0 && probability < 1, "The sampling rate must be greater than 0.0 and less than 1.0")
+        self.probability = probability
+        self.lowUpperBound = UInt64(probability * Double(UInt64.max))
+    }
+
+    public func sample(operationName: String, traceID: TraceID) -> SamplingStatus {
+        SamplingStatus(isSampled: traceID.low < self.lowUpperBound, attributes: [
```

Comment:
oh i see... interesting, so we use the fact that low must be randomly generated...

Is it always? I know some tracers sometimes use a few bytes in the IDs to carry additional information?